Source/WebKit/UIProcess/PeriodicalResponsivenessTimer.cpp

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2022 Igalia S.L.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "PeriodicalResponsivenessTimer.h"
+
+#include "WebPageProxy.h"
+#include "WebProcessProxy.h"
+
+namespace WebKit {
+
+PeriodicalResponsivenessTimer::PeriodicalResponsivenessTimer(WebPageProxy& page)
+    : m_page(page)
+    , m_timer(RunLoop::main(), this, &PeriodicalResponsivenessTimer::timerFired)
+{
+    start();
+}
+
+PeriodicalResponsivenessTimer::~PeriodicalResponsivenessTimer()
+{
+    stop();
+}
+
+void PeriodicalResponsivenessTimer::start()
+{
+    timerFired();
+    m_timer.startRepeating(m_interval);
+}
+
+void PeriodicalResponsivenessTimer::stop()
+{
+    m_timer.stop();
+}
+
+void PeriodicalResponsivenessTimer::timerFired()
+{
+    if (m_page.hasRunningProcess())
+        m_page.process().isResponsive(nullptr);
```

Comment:
It's interesting that you don't have a callback here to see whether the web process is responsive or not. I guess it works by triggering all other callbacks to execute? This merits a comment to explain what you're doing, since it's not obvious.

Reply:
yes, it is exactly as you comment. The consequence of the process().isResponsive(nullptr); is the ResponsivenessTimer starts and after 3 sec it triggers all other callbacks are called and signal notify::is-web-process-responsive is sent. I will add a comment about that.

Reply:
To do this I would need to change the ResponsivenessTimer, if I change the interval of PeriodicalResponsivenessTimer I change only how often the ResponsivenessTimer will start, when it ends(3sec, then the notify::is-web-process-responsive is sent)  is something separate.